src/mono/mono/metadata/class-internals.h

```diff
@@ -1003,7 +1003,9 @@ typedef struct {
 	MonoClass *iremotingtypeinfo_class;
 #endif
 	MonoClass *mono_method_message_class;
+#ifndef ENABLE_NETCORE
 	MonoClass *appdomain_class;
```

Comment:
There is one more use of this in mono/metadata/runtime.c - `fire_process_exit_event`

Additionally, there is a use of `mono_class_get_appdomain_class` `mono_runtime_init_checked`

And uses of `mono_class_get_appdomain_setup_class` in
* `mono_runtime_init_checked`
* `mono_domain_create_appdomain_checked` and `copy_app_domain_setup` (these should probably go under an ifdef for netcore)

Also I would change the `GENERATE_GET_CLASS_WITH_CACHE (appdomain, MONO_APPDOMAIN_CLASS_NAME_SPACE, MONO_APPDOMAIN_CLASS_NAME);` in `appdomain.c` and `GENERATE_GET_CLASS_WITH_CACHE_DECL (appdomain)` in `class-internals.h` (and the `appdomain_setup`, too) to ifdef out on netcore.

Reply:
> There is one more use of this in mono/metadata/runtime.c - fire_process_exit_event

It uses `appcontext_class` in netcore so that's fine

> Additionally, there is a use of mono_class_get_appdomain_class in mono_runtime_init_checked

Yeah, I had to undo the managed types removal, there is still tons of code that depends on it, and instead of if-defing all this out we should just delete the code.